### PR TITLE
Allow port configuration, refactor start actor modal

### PR DESCRIPTION
**Pull request by brooksmtownsend**
This started as just a port configuration PR, but with some more UI changes with actors I figured I'd give refactoring the modal a try.

Along with port configuration, this PR also shows the appropriate host ID in the actor table, and the new "homepage" supposed to be the mockup for the full lattice view. In subsequent PRs, the lattice view will show a list of hosts that can be further inspected for only the resources (actors, providers) running on that specific host.

What I'd love feedback on is the code for the start actor component, how it's laid out in the page_live template, and how it looks when being used. Refactoring to live components makes it so that the modal disappears and only the table re-renders on the page, instead of a full POST request + page reload. I'd like to do the same thing to the define-link modal and the start providers modals, but I'd like some feedback before putting in the effort (and potentially mistakes) on those.

### CODE REVIEW

File: wasmcloud_host/assets/vendor/wasmcloud/js/wasmcloud.js [hunk 1]
```diff
@@ -1,173 +1,173 @@
...
-            },
-            point: {
-                radius: 4,
-                hitRadius: 10,
-                hoverRadius: 4
-            }
-        }
-    }
-}); // eslint-disable-next-line no-unused-vars
+// // Custom JS for wasmCloud logic goes here
```
Comment (autodidaddict) [inline wasmcloud_host/assets/vendor/wasmcloud/js/wasmcloud.js:1]: Once this PR isn't a WIP, please remove all commented out code.

File: wasmcloud_host/config/config.exs [hunk 2]
```diff
@@ -23,6 +23,11 @@ config :logger, :console,
 # Use Jason for JSON parsing in Phoenix
 config :phoenix, :json_library, Jason
 
+# Allow wasm uploads
+config :mime, :types, %{
```
Comment (autodidaddict) [inline wasmcloud_host/config/config.exs:27]: :+1:

File: wasmcloud_host/lib/wasmcloud_host_web/live/page_live.html.leex [hunk 3]
```diff
@@ -112,98 +112,108 @@
...
+    <div class="card">
+      <div class="card-header">
+        <div class="row justify-content-between">
+          <div class="col-lg-2 col-md-3 col-sm-3" style="max-width:150px">
+            <h3>Providers</h3>
+          </div>
+          <div class="mfe-2">
+            <div class="dropdown d-inline-block">
+              <button class="btn btn-secondary dropdown-toggle" id="dropdownMenuButton" type="button" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">Start Provider</button>
+              <div class="dropdown-menu" aria-labelledby="dropdownMenuButton">
```
Comment (autodidaddict) [inline wasmcloud_host/lib/wasmcloud_host_web/live/page_live.html.leex:167]: Is this `aria` stuff specific to Phoenix, or is this some front-end thing that I just missed out on while working on backends for the past 5 years?

File: wasmcloud_host/lib/wasmcloud_host_web/live/page_live.html.leex [hunk 4]
```diff
@@ -261,3 +271,30 @@
...
+<%# Start actor modal (One for file, one for OCI) %>
+<div class="modal fade" id="start_actor_file" tabindex="-1" role="dialog" aria-labelledby="start_actor_file" aria-hidden="true">
+  <div class="modal-dialog" role="document">
+    <div class="modal-content">
+      <div class="modal-header">
+        <h4 class="modal-title">Start actor from file</h4>
+        <button class="close" type="button" data-dismiss="modal" aria-label="Close"><span aria-hidden="true">×</span></button>
+      </div>
+      <div class="modal-body">
+        <%= live_component StartActorComponent, id: :start_actor_file_modal %>
```
Comment (autodidaddict) [inline wasmcloud_host/lib/wasmcloud_host_web/live/page_live.html.leex:283]: :+1:

File: wasmcloud_host/lib/wasmcloud_host_web/live/start_actor_component.ex [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+        {:ok, bytes} = File.read(path)
+        replicas |> Enum.each(fn _ -> HostCore.Actors.ActorSupervisor.start_actor(bytes) end)
+      end)
+
+    {:noreply, socket}
+  end
+
+  def handle_event(
+        "start_actor_ociref",
+        %{"_csrf_token" => _csrf_token, "replicas" => replicas, "actor_ociref" => actor_ociref},
```
Comment (autodidaddict) [inline wasmcloud_host/lib/wasmcloud_host_web/live/start_actor_component.ex:31]: You should be able to simply leave off the `"_csrf_token"` part of the match. The match should still pass while you're ignoring the CSRF token.

File: wasmcloud_host/assets/vendor/wasmcloud/js/wasmcloud.js [hunk 6]
```diff
@@ -1,173 +1,173 @@
...
- * Thanks, JavaScript.
- */
-null.ignoreThisError
\ No newline at end of file
+// /**
+//  * For some reason, stopping the execution of this file here makes the charts load correctly.
+//  * Until we figure that out, leave this in.
+//  * Thanks, JavaScript.
+//  */
+// null.ignoreThisError
```
Comment (brooksmtownsend) [inline wasmcloud_host/assets/vendor/wasmcloud/js/wasmcloud.js]: As fun as this hack was, it does break the live view reloading, so the charts are going to have to be less impressive for a short time. They still render, just in a way that seems a little incorrect, so we can open up an issue to address that once we tackle #45

File: wasmcloud_host/lib/wasmcloud_host_web/live/start_actor_component.ex [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+      <div class="form-group row">
+        <label class="col-md-3 col-form-label" for="text-input">Replicas</label>
+        <div class="col-md-9">
+          <input class="form-control" id="number-input" type="number" name="replicas" placeholder="1" value="1"><span class="help-block">Enter how many instances of this actor you want</span>
+        </div>
+      </div>
+      <div class="modal-footer">
+        <button id="close_modal-<%= @id %>" class="btn btn-secondary" type="button" data-dismiss="modal">Close</button>
+        <!-- onClick event closes modal -->
+        <button class="btn btn-primary" type="submit" onClick="document.getElementById('close_modal-<%= @id %>').click()">Submit</button>
```
Comment (autodidaddict) [inline wasmcloud_host/lib/wasmcloud_host_web/live/start_actor_component.ex:78]: Is this `close_modal-{id}` thing part of Phoenix or something you wrote on your own?

File: wasmcloud_host/lib/wasmcloud_host_web/templates/layout/root.html.leex [hunk 8]
```diff
@@ -92,7 +92,7 @@
                 Version<span class="badge badge-success mfs-auto">0.19.0</span></a><a class="dropdown-item" href="#">
                 <svg class="c-icon mfe-2">
                   <use xlink:href="<%= Routes.static_path(@conn, "/coreui/free.svg#cil-account-logout") %>"></use>
-                </svg> Logout</a></div>
+                </svg> Eject</a></div>
```
Comment (autodidaddict) [inline wasmcloud_host/lib/wasmcloud_host_web/templates/layout/root.html.leex:95]: :smile:

**Review by autodidaddict — CHANGES_REQUESTED** [review]
Only changes I suggest is removing the commented out JavaScript.

File: wasmcloud_host/lib/wasmcloud_host_web/live/page_live.html.leex (hunk 3 above)
Comment (brooksmtownsend) [reply wasmcloud_host/lib/wasmcloud_host_web/live/page_live.html.leex:167]: `aria` is a set of html attributes that can help make websites more accessible, and a lot of it came prefilled from the coreUI templates. I'm not 100% sure how it ends up interacting with accessibility tools, though.

File: wasmcloud_host/lib/wasmcloud_host_web/live/start_actor_component.ex (hunk 7 above)
Comment (brooksmtownsend) [reply wasmcloud_host/lib/wasmcloud_host_web/live/start_actor_component.ex:78]: I stole that from https://hexdocs.pm/phoenix_live_view/Phoenix.LiveComponent.html#module-stateful-components-life-cycle, which was a way to set a unique DOM ID from the component id. It's a really similar pattern to what I've done before in React, where each dom element in certain scenarios needs a unique ID

File: wasmcloud_host/lib/wasmcloud_host_web/live/start_actor_component.ex (hunk 5 above)
Comment (brooksmtownsend) [reply wasmcloud_host/lib/wasmcloud_host_web/live/start_actor_component.ex:31]: Nice, I'll give that a shot

File: wasmcloud_host/lib/wasmcloud_host/lattice/state_monitor.ex [hunk 9]
```diff
@@ -195,6 +195,48 @@ defmodule WasmcloudHost.Lattice.StateMonitor do
...
+  defp process_event(
+         state,
+         %Cloudevents.Format.V_1_0.Event{
+           data: %{"public_key" => _pkey},
+           datacontenttype: "application/json",
+           source: _source_host,
+           type: "com.wasmcloud.lattice.health_check_passed"
+         }
+       ) do
+    Logger.info("Handling heartbeat success")
```
Comment (autodidaddict) [inline wasmcloud_host/lib/wasmcloud_host/lattice/state_monitor.ex:221]: I know that this is just a placeholder, but this isn't a heartbeat, this is a health check.

File: wasmcloud_host/lib/wasmcloud_host/lattice/state_monitor.ex [hunk 10]
```diff
@@ -195,6 +195,48 @@ defmodule WasmcloudHost.Lattice.StateMonitor do
...
+  defp process_event(
+         state,
+         %Cloudevents.Format.V_1_0.Event{
+           data: %{"public_key" => _pkey},
+           datacontenttype: "application/json",
+           source: _source_host,
+           type: "com.wasmcloud.lattice.health_check_failed"
+         }
+       ) do
+    Logger.info("Handling heartbeat failure")
```
Comment (autodidaddict) [inline wasmcloud_host/lib/wasmcloud_host/lattice/state_monitor.ex:235]: health check ;)

**Review by autodidaddict — CHANGES_REQUESTED** [review]
I know they're placeholders, but just change the info logger for handling the health check pass/fail.

**Review by autodidaddict — APPROVED** [review]

**Pull request merged**